### PR TITLE
Prepare NuGet.Jobs to update client libraries

### PR DESCRIPTION
**Pull request by loic-sharma**
TLDR: Currently the `Validation.PackageSigning.Core.Tests` project is used as a common helper project by other test projects. This change moves as much of the common helpers to our `TestUtil` project.

<details>
<summary>Longer version...</summary>

The `Validation.PackageSigning.Core.Tests` project references client's `Test.Utility` package which we will need to update. The client's `Test.Utility` package causes massive binding redirect headaches as it references Visual Studio DLLs. To reduce the blast radius of these binding redirects, I reduced the number of projects that depend on `Validation.PackageSigning.Core.Tests`. This will make fixing the binding redirect issues *much* more manageable.

You can find my prototype of updating to client's private release here: https://github.com/NuGet/NuGet.Jobs/compare/dev...loshar-skip-sha1
</details>

Part of https://github.com/NuGet/NuGetGallery/issues/8197

### CODE REVIEW

File: tests/StatusAggregator.Tests/Collector/ManualStatusChangeCollectorProcessorTests.cs [hunk 1]
```diff
@@ -90,7 +90,7 @@ public async Task HandlesManualChanges()
 
         public class TheFetchSinceMethodAtMinValue : TheFetchSinceMethod
         {
-            public override DateTime Cursor => DateTime.MinValue.ToUniversalTime();
+            public override DateTime Cursor => DateTime.MinValue;
```
Comment (loic-sharma) [inline tests/StatusAggregator.Tests/Collector/ManualStatusChangeCollectorProcessorTests.cs:93]: This test is currently failing. The job uses `DateTime.MinValue` as its default: https://github.com/NuGet/NuGet.Jobs/blob/cb8e8996f0d85272bcaa812f14a61c1bd81bb495/src/StatusAggregator/Collector/Cursor.cs#L38

File: tests/NuGet.Jobs.GitHubIndexer.Tests/GitHubSearchWrapperFacts.cs [hunk 2]
```diff
@@ -35,7 +35,7 @@ private static GitHubSearchWrapper GetTestSearcher(IReadOnlyDictionary<string, s
 
             mockConnection
                 .Setup(x => x.Get<SearchRepositoryResult>(It.IsAny<Uri>(), It.IsAny<IDictionary<string, string>>(), It.IsAny<string>()))
-                .Returns(async (Uri uri, IDictionary<string, string> parameters, string accepts) =>
+                .ReturnsAsync((Uri uri, IDictionary<string, string> parameters, string accepts) =>
```
Comment (loic-sharma) [inline tests/NuGet.Jobs.GitHubIndexer.Tests/GitHubSearchWrapperFacts.cs:38]: xunit warns about this

File: tests/StatusAggregator.Tests/Collector/ManualStatusChangeCollectorProcessorTests.cs (hunk 1 above)
Comment (joelverhagen) [reply tests/StatusAggregator.Tests/Collector/ManualStatusChangeCollectorProcessorTests.cs:93]: Please test this in Belgium time 😄

File: tests/TestUtil/TestUtil.csproj [hunk 3]
```diff
@@ -41,15 +41,33 @@
...
   <ItemGroup>
+    <PackageReference Include="Microsoft.Extensions.Logging.Abstractions">
+      <Version>2.2.0</Version>
+    </PackageReference>
     <PackageReference Include="xunit">
       <Version>2.4.1</Version>
     </PackageReference>
   </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\..\src\Validation.Common.Job\Validation.Common.Job.csproj">
```
Comment (joelverhagen) [inline tests/TestUtil/TestUtil.csproj:63]: This seems to specific for a generic "TestUtil" right? Should this project really be `Validation.TestUtil` or something?

I don't feel strongly.

**Review by joelverhagen — APPROVED** [review]

File: tests/TestUtil/TestUtil.csproj (hunk 3 above)
Comment (loic-sharma) [reply tests/TestUtil/TestUtil.csproj:63]: Good catch! I believe this isn't necessary. I copied this from `Validation.PackageSigning.Core.Tests.csproj` when I moved the helpers down. I'll try to remove this reference

File: tests/StatusAggregator.Tests/Collector/ManualStatusChangeCollectorProcessorTests.cs (hunk 1 above)
Comment (loic-sharma) [reply tests/StatusAggregator.Tests/Collector/ManualStatusChangeCollectorProcessorTests.cs:93]: Good idea. This caught more test failures that I've now fixed
